pkg/nfd-worker/nfd-worker.go

```diff
@@ -271,6 +314,43 @@ func (w *nfdWorker) Run() error {
 	labelTrigger.Reset(w.config.Core.SleepInterval.Duration)
 	defer labelTrigger.Stop()
 
+	// Create owner ref
+	ownerReference := []metav1.OwnerReference{}
+	// Get pod owner reference
+	podName := os.Getenv("POD_NAME")
+
+	selfPod, err := w.k8sClient.CoreV1().Pods(w.kubernetesNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
```

Comment:
being able to get the owning daemonset to be later set for NodeFeature CR is critical IMO and should not be treated a best effort. especially when we attempt to do it once at start. 

can we fail Run() if error occurred but is not of type IsNotFound ? (which AFAIU is to support dev mode)

personally i prefer to be strict and always fail here and below (when getting pod). adding an option to skip failure via env var or flag (e.g DEV_MODE or --dev) so we have a deterministic behaviour.

Reply:
@marquiz ?